### PR TITLE
Enable path-label munger

### PR DESCRIPTION
**Pull request by 0xmichalis**
@stevekuznetsov @deads2k

Tested in https://github.com/origin-prow/origin/pull/19

### CODE REVIEW

File: cluster/ci/config/submit-queue/submit_queue.yaml [hunk 1]
```diff
@@ -254,10 +259,30 @@ objects:
...
+    name: path-label
+  data:
+    path-label.txt: |-
+      # This file is used by the path-label munger and is of the form:
+      # PATH REGEXP     LABEL
+
+      # examples:
+      # pkg/api/types.go
+      # pkg/api/*/types.go
+      ^pkg/api/([^/]+/)?types.go$   needs-api-review
```
Comment (deads2k) [inline cluster/ci/config/submit-queue/submit_queue.yaml]: expand this to catch anything with the path at all so we catch naughty upstream picks too?

File: cluster/ci/config/submit-queue/submit_queue.yaml [hunk 2]
```diff
@@ -254,10 +259,30 @@ objects:
...
+
+      # examples:
+      # pkg/*/apis/*/types.go
+      # pkg/*/apis/*/*/types.go
+      ^pkg/([^/]+/)*apis/[^/]+/([^/]+/)?types.go$   needs-api-review
```
Comment (deads2k) [inline cluster/ci/config/submit-queue/submit_queue.yaml]: expand this to catch anything with the path at all so we catch naughty upstream picks too?

File: cluster/ci/config/submit-queue/submit_queue.yaml (hunk 1 above)
Comment (kargakis) [inline cluster/ci/config/submit-queue/submit_queue.yaml]: Do you want a separate regexp for vendor or are you thinking of matching any "types.go" file in the repo?

Comment (deads2k) [inline cluster/ci/config/submit-queue/submit_queue.yaml]: > Do you want a separate regexp for vendor or are you thinking of matching any "types.go" file in the repo?

Probably `*/[api or apis]/*types.go`, so we don't have predict all the vendor paths.

Comment (kargakis) [inline cluster/ci/config/submit-queue/submit_queue.yaml]: updated, ptal while I am testing this

Comment (kargakis) [inline cluster/ci/config/submit-queue/submit_queue.yaml]: PR changing upstream code: https://github.com/origin-prow/origin/pull/21

File: cluster/ci/config/submit-queue/submit_queue.yaml [hunk 3]
```diff
@@ -255,10 +260,30 @@ objects:
...
+    name: path-label
+  data:
+    path-label.txt: |-
+      # This file is used by the path-label munger and is of the form:
+      # PATH REGEXP     LABEL
+
+      # examples:
+      # pkg/api/types.go
+      # pkg/api/*/types.go
+      .*/api/([^/]+/)*types.go$   needs-api-review
```
Comment (deads2k) [inline cluster/ci/config/submit-queue/submit_queue.yaml]: this lgtm

**Comment by kargakis** [comment]
Deployed.

**Pull request merged**